src/model/utils/errorResponse.ts

```diff
@@ -1,3 +1,9 @@
+import { MessageField } from "./messageField";
+
 export class ErrorResponse {
-    constructor(public error?: string, public message?: string){}
+    constructor(public error?: string, public message?: string, public detail?: Array<MessageField>){};
+
+    public setDetail(aDetail: Array<MessageField>): void {
```

Comment:
Nice, would like to do this more, 
The trade-off of typing and Maintaining the settera and getters are a bit of a pain for new system.

In Java projects there is a library called lambok which generated the getters and setters for classes

Would be good to find a way in the future to find an npm lib that creates getters and setters when reading preprocessor